schema/collection.go

```diff
@@ -100,6 +102,11 @@ func NewDefaultCollection(name string, id uint32, schVer int, ctype CollectionTy
 
 	queryableFields := BuildQueryableFields(fields)
 
+	partitionFields, err := BuildPartitionFields(schema, fields)
```

Comment:
Can this be built during schema parsing similar to how we set the primary key in the fields? Basically in `deserializeProperties` method we can pass this and set this attribute in the field